ivy/ivy_solver.py

```diff
@@ -603,18 +603,20 @@ def formula_to_z3_int(fmla):
         return z3.Or(args)
     if isinstance(fmla,ivy_logic.Not):
         return z3.Not(args[0])
-    if isinstance(fmla,ivy_logic.Definition or ivy_logic.is_eq(fmla) or isinstance(fmla,ivy_logic.Iff)):
-        if ivy_logic.is_true(fmla.args[1]):
-            return args[0]
-        if ivy_logic.is_false(fmla.args[1]):
-            return z3.Not(args[0])
+    if isinstance(fmla,ivy_logic.Definition or ivy_logic.is_eq(fmla)):
```

Comment:
There seem to remain a parenthesis issue here.

Reply:
I've fixed this in a new commit.